src/distribute_kv_cache/rpc/message.rs

```diff
@@ -0,0 +1,1224 @@
+        let status = match buf.get(40) {
+            Some(&0) => StatusCode::Success,
+            Some(&1) => StatusCode::NotFound,
+            Some(&2) => StatusCode::InternalError,
+            Some(&3) => StatusCode::VersionMismatch,
+            _ => return Err(RpcError::InternalError("Invalid status code".to_owned())),
+        };
+        let data = buf.get(41..).unwrap_or(&[]).to_vec();
+        let data_len = usize_to_u64(data.len());
+        if data_len != block_size {
+            return Err(RpcError::InternalError(format!(
+                "Insufficient block size {data_len}"
+            )));
+        }
+
+        Ok(FileBlockResponse {
+            file_id,
+            block_id,
+            block_size,
+            block_version,
+            hash_ring_version,
+            status,
+            data,
+        })
+    }
+}
+
+/// The status code of the response.
+#[derive(
+    Debug, IntoPrimitive, TryFromPrimitive, PartialEq, Eq, Clone, Copy, Serialize, Deserialize,
+)]
+#[repr(u8)]
+pub enum StatusCode {
+    /// The request is successful.
+    Success = 0,
+    /// The request is not found.
+    NotFound = 1,
+    /// The request is invalid.
+    InternalError = 2,
+    /// The request is out dated.
+    VersionMismatch = 3,
+}
+
+impl Default for StatusCode {
+    /// Get the default status code.
+    fn default() -> Self {
+        Self::Success
+    }
+}
+
+/// The file block packet for client
+/// This struct impl packet trait and support file request and response
+#[derive(Debug, Clone)]
+pub struct FileBlockRequestTask {
+    /// The sequence number of the packet.
+    pub seq: u64,
+    /// The operation type of the packet.
+    pub op: u8,
+    /// The timestamp of the packet.
+    pub timestamp: u64,
+    /// The file block request struct.
+    pub request: FileBlockRequest,
+    /// The buffer of the packet, used to store response data.
+    pub response: Option<FileBlockResponse>,
+    /// The sender to send response back to caller.
+    pub done_tx: flume::Sender<Result<FileBlockResponse, FileBlockRequest>>,
+}
+
+impl FileBlockRequestTask {
+    /// Create a new file block packet.
+    #[must_use]
+    pub fn new(
+        block_request: &FileBlockRequest,
+        done_tx: flume::Sender<Result<FileBlockResponse, FileBlockRequest>>,
+    ) -> Self {
+        Self {
+            // Will be auto set by client
+            seq: 0,
+            // Set operation
+            op: ReqType::FileBlockRequest.into(),
+            request: block_request.clone(),
+            timestamp: 0,
+            response: None,
+            done_tx,
+        }
+    }
+}
+
+impl Encode for FileBlockRequestTask {
+    /// Encode the file block packet into a byte buffer.
+    fn encode(&self, buffer: &mut BytesMut) {
+        let mut writer = buffer.writer();
+        if let Err(err) = bincode::serialize_into(&mut writer, &self.request) {
+            error!("Failed to serialize: {}", err);
+        }
+    }
+}
```

Comment:
If all encode implementations are the same, you can create macro to avoid repeating.

Reply:
Yes, I have updated it in previous commit.